File: ccif/src/java/other/org/himalay/ccif/programCnfMsg.java
// Copyright (2013) Krishna C Tripathi. All rights reserved.
// 
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
package org.himalay.ccif ;

import java.io.DataInputStream;
import java.io.DataOutputStream;
import java.io.IOException;
import java.lang.Exception;
import java.io.ByteArrayInputStream;
import java.io.PrintStream;
import org.himalay.msgs.runtime.Created;

import org.himalay.msgs.runtime.*;
@Created(date = "Fri Jan 15 01:32:55 EST 2016")

public  class programCnfMsg extends BinStruct implements PublicBinMsg{




    // members variables
    // purchaseType
    public short purchaseType ;
    // purchasePrice
    public int purchasePrice ;
    // purchaseValidation
    public short purchaseValidation ;
    // expirationdatee
    public long expirationdatee ;
    // ProgramStartTim
    public long ProgramStartTim ;
    // initialFreepreviewDuration
    public int initialFreepreviewDuration ;
    // anyTimeFreepreviewDuration
    public int anyTimeFreepreviewDuration ;
    // data
    public ByteArray data ;
    // data1
    public ByteArray data1 ;
    // descriptors
    public ArrayList<CaDescriptor> descriptors ;

        public programCnfMsg () // throws Exception
    {
            init();
    }
    
    private void init()
    {
        // Initialize purchaseType
        
        // Initialize purchasePrice
        
        // Initialize purchaseValidation
        
        // Initialize expirationdatee
        
        // Initialize ProgramStartTim
        
        // Initialize initialFreepreviewDuration
        
        // Initialize anyTimeFreepreviewDuration
        
        // Initialize data
        data= new ByteArray();data.setSizeType("FIRST_UI8");
        // Initialize data1
        data1= new ByteArray();data1.setSizeType("FIRST_UI8");
        // Initialize descriptors
        descriptors= new ArrayList<CaDescriptor>();descriptors.setMemberSize(0);
    }

    public int readNoHeader(DataInputStream istream) throws IOException 
    {
    
     return read(istream);
       }
   
    public int read(DataInputStream istream) throws IOException 
    {
    preRead();
        int retVal= 0;

                     // read purchaseType
        {purchaseType=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read purchasePrice
        {purchasePrice=istream.readUnsignedShort(); retVal+=2;}
        // read purchaseValidation
        {purchaseValidation=(short)(istream.readUnsignedByte()); retVal+=1;}
        // read expirationdatee
        {expirationdatee=(long)(BinPrimitive.readUI32(istream) ); retVal+=4;}
        // read ProgramStartTim
        {ProgramStartTim=(long)(BinPrimitive.readUI32(istream) ); retVal+=4;}
        // read initialFreepreviewDuration
        {initialFreepreviewDuration=istream.readUnsignedShort(); retVal+=2;}
        // read anyTimeFreepreviewDuration
        {anyTimeFreepreviewDuration=istream.readUnsignedShort(); retVal+=2;}
        // read data
        {retVal+=data.read(istream); }
        // read data1
        {retVal+=data1.read(istream); }
        // read descriptors
        for (; istream.available() > 0 ; ){     CaDescriptor temp;    temp = new CaDescriptor();    retVal += temp.read(istream);    descriptors.add(temp);}

        postRead();
        return retVal;
    }


    public int write(DataOutputStream ostream) throws IOException 
    {
    preWrite();
        int retVal= 0;

        
        
        
        
        
        
        
        { /** fix dependent sizes for data**/  }
        { /** fix dependent sizes for data1**/  }
        { /** fix dependent sizes for descriptors**/  }
    
                // write purchaseType
        ostream.writeByte(purchaseType); retVal +=1;
        // write purchasePrice
        ostream.writeShort(purchasePrice); retVal +=2;
        // write purchaseValidation
        ostream.writeByte(purchaseValidation); retVal +=1;
        // write expirationdatee
        BinPrimitive.writeUI32(ostream,expirationdatee); retVal +=4;
        // write ProgramStartTim
        BinPrimitive.writeUI32(ostream,ProgramStartTim); retVal +=4;
        // write initialFreepreviewDuration
        ostream.writeShort(initialFreepreviewDuration); retVal +=2;
        // write anyTimeFreepreviewDuration
        ostream.writeShort(anyTimeFreepreviewDuration); retVal +=2;
        // write data
        {retVal += data.write(ostream);}
        // write data1
        {retVal += data1.write(ostream);}
        // write descriptors
        {ArrayList<CaDescriptor> temp1 = descriptors;for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++){     CaDescriptor temp2    =    temp1.get(iIdx);    if (temp2!=null)retVal +=temp2.write(ostream);}}
postWrite();
        return retVal;
    }
    
    public int dump(DumpContext dc) throws IOException 
    {
        dc.indent();dc.getPs().print("programCnfMsg\n");
    dc.increaseIndent();
        int retVal= 0;
        // write purchaseType
        dc.indent();dc.getPs().println("purchaseType="+purchaseType+"(0x"+ Integer.toHexString(purchaseType) + ")" );
        // write purchasePrice
        dc.indent();dc.getPs().println("purchasePrice="+purchasePrice+"(0x"+ Integer.toHexString(purchasePrice) + ")" );
        // write purchaseValidation
        dc.indent();dc.getPs().println("purchaseValidation="+purchaseValidation+"(0x"+ Integer.toHexString(purchaseValidation) + ")" );
        // write expirationdatee
        dc.indent();dc.getPs().println("expirationdatee="+expirationdatee+"(0x"+ Long.toHexString(expirationdatee)+")") ;
        // write ProgramStartTim
        dc.indent();dc.getPs().println("ProgramStartTim="+ProgramStartTim+"(0x"+ Long.toHexString(ProgramStartTim)+")") ;
        // write initialFreepreviewDuration
        dc.indent();dc.getPs().println("initialFreepreviewDuration="+initialFreepreviewDuration+"(0x"+ Integer.toHexString(initialFreepreviewDuration) + ")" );
        // write anyTimeFreepreviewDuration
        dc.indent();dc.getPs().println("anyTimeFreepreviewDuration="+anyTimeFreepreviewDuration+"(0x"+ Integer.toHexString(anyTimeFreepreviewDuration) + ")" );
        // write data
        dc.indent();dc.getPs().print("data: "+data.getSize()+"(0x"+Integer.toHexString(data.getSize())+")\n");this.data.dump(dc);
        // write data1
        dc.indent();dc.getPs().print("data1: "+data1.getSize()+"(0x"+Integer.toHexString(data1.getSize())+")\n");this.data1.dump(dc);
        // write descriptors
        { ArrayList<CaDescriptor> temp1 = descriptors;for (int iIdx=0; iIdx < temp1.getCount() ; iIdx++){     CaDescriptor element    = temp1.get(iIdx); dc.indent(); dc.getPs().println(iIdx);    if ( element != null ) {dc.indent();dc.getPs().println("element") ;retVal +=element.dump(dc);}}}
dc.decreaseIndent();
        return retVal;
    }


        // Getter for purchaseType
    //public short getPurchaseType()
    //{
    //    return purchaseType ;
    //}

    
    // Setter for purchaseType
    //public void setPurchaseType(short val)
    //{
    //    this.purchaseType= val;
    //}
    // Getter for purchasePrice
    //public int getPurchasePrice()
    //{
    //    return purchasePrice ;
    //}

    
    // Setter for purchasePrice
    //public void setPurchasePrice(int val)
    //{
    //    this.purchasePrice= val;
    //}
    // Getter for purchaseValidation
    //public short getPurchaseValidation()
    //{
    //    return purchaseValidation ;
    //}

    
    // Setter for purchaseValidation
    //public void setPurchaseValidation(short val)
    //{
    //    this.purchaseValidation= val;
    //}
    // Getter for expirationdatee
    //public long getExpirationdatee()
    //{
    //    return expirationdatee ;
    //}

    
    // Setter for expirationdatee
    //public void setExpirationdatee(long val)
    //{
    //    this.expirationdatee= val;
    //}
    // Getter for ProgramStartTim
    //public long getProgramStartTim()
    //{
    //    return ProgramStartTim ;
    //}

    
    // Setter for ProgramStartTim
    //public void setProgramStartTim(long val)
    //{
    //    this.ProgramStartTim= val;
    //}
    // Getter for initialFreepreviewDuration
    //public int getInitialFreepreviewDuration()
    //{
    //    return initialFreepreviewDuration ;
    //}

    
    // Setter for initialFreepreviewDuration
    //public void setInitialFreepreviewDuration(int val)
    //{
    //    this.initialFreepreviewDuration= val;
    //}
    // Getter for anyTimeFreepreviewDuration
    //public int getAnyTimeFreepreviewDuration()
    //{
    //    return anyTimeFreepreviewDuration ;
    //}

    
    // Setter for anyTimeFreepreviewDuration
    //public void setAnyTimeFreepreviewDuration(int val)
    //{
    //    this.anyTimeFreepreviewDuration= val;
    //}
    // Getter for data
    //public ByteArray getData()
    //{
    //    return data ;
    //}

    
    // Setter for data
    //public void setData(ByteArray val)
    //{
    //    this.data= val;
    //}
    // Getter for data1
    //public ByteArray getData1()
    //{
    //    return data1 ;
    //}

    
    // Setter for data1
    //public void setData1(ByteArray val)
    //{
    //    this.data1= val;
    //}
    // Getter for descriptors
    //public ArrayList<CaDescriptor> getDescriptors()
    //{
    //    return descriptors ;
    //}

    
    // Setter for descriptors
    //public void setDescriptors(ArrayList<CaDescriptor> val)
    //{
    //    this.descriptors= val;
    //}


    public void setData(byte[] val)
    {
        this.data.setData(val);
    }
    
    
    public void setData1(byte[] val)
    {
        this.data1.setData(val);
    }
    
    
    public int addToDescriptors(CaDescriptor val)
    {
            descriptors.add(val); return descriptors.size();
    }
    
    
    public int removeFromDescriptors(CaDescriptor val)
    {
            descriptors.remove(val); return descriptors.size();
    }
    
    
    public int removeNthFromDescriptors(int idx)
    {
            descriptors.remove(idx); return descriptors.size();
    }
    
    
    public int emptyDescriptors(int idx)
    {
            descriptors.clear(); return descriptors.size();
    }
    
    
    public int getSize() throws IOException
    {
       DataOutputStream dos= new DataOutputStream(new NullStream());
       return this.write(dos);
    }
    
        

}

// End of code

File: libccif/src/bp/programCnfMsg.cpp
//**//
//@Created(date = "Mon Feb 01 22:30:13 EST 2016")
// Copyright (2012) Krishna C Tripathi. All rights reserved.
//
// You are not allowed to read/copy/distribute following code without explicit written authorization from Krishna C Tripathi
//
//**//
#include "stdio.h"
#include "programCnfMsg.h"

#ifdef org_himalay_ccif_programCnfMsg__USE_SMART_PTR
#define ArrayList(x) SmartPtrList<x>
#define ArrayList_iterator(x) SmartPtrList<x>::iterator
#include "SmartPtrList.h"
#else
#define ArrayList(x) BinMessagePtrList
#define ArrayList_iterator(x) BinMessagePtrList::iterator
#include "BinMessagePtrList.h"
#endif

// Namespace
namespace org {
namespace himalay {
namespace ccif {

using namespace ::org::himalay::msgs::runtime;

// Constructor
programCnfMsg::programCnfMsg() {
    references= 0;


    // purchaseType
    purchaseType= (ui8)0;
    // purchasePrice
    purchasePrice= (ui16)0;
    // purchaseValidation
    purchaseValidation= (ui8)0;
    // expirationdatee
    expirationdatee= (ui32)0;
    // ProgramStartTim
    ProgramStartTim= (ui32)0;
    // initialFreepreviewDuration
    initialFreepreviewDuration= (ui16)0;
    // anyTimeFreepreviewDuration
    anyTimeFreepreviewDuration= (ui16)0;
    // data
    org_himalay_msgs_runtime_ByteArray_NEW(data);
    data->setSizeType("FIRST_UI8");
    // data1
    org_himalay_msgs_runtime_ByteArray_NEW(data1);
    data1->setSizeType("FIRST_UI8");
    // descriptors
    descriptors.setMemberSize(0);
}
// Destructor
programCnfMsg::~programCnfMsg() {
    // purchaseType

    // purchasePrice

    // purchaseValidation

    // expirationdatee

    // ProgramStartTim

    // initialFreepreviewDuration

    // anyTimeFreepreviewDuration

    // data
    org_himalay_msgs_runtime_ByteArray_DELETE( data);
    // data1
    org_himalay_msgs_runtime_ByteArray_DELETE( data1);
    // descriptors
    {
#ifndef org_himalay_ccif_CaDescriptor_USE_SMART_PTR
        descriptors.deleteMembers();
#endif
    }
}

int programCnfMsg::readNoHeader(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;

    // purchaseType
    {
        purchaseType=(istream.readUI8());
        retVal+=1;
    }
    // purchasePrice
    {
        purchasePrice=istream.readUI16();
        retVal+=2;
    }
    // purchaseValidation
    {
        purchaseValidation=(istream.readUI8());
        retVal+=1;
    }
    // expirationdatee
    {
        expirationdatee=istream.readUI32() ;
        retVal+=4;
    }
    // ProgramStartTim
    {
        ProgramStartTim=istream.readUI32() ;
        retVal+=4;
    }
    // initialFreepreviewDuration
    {
        initialFreepreviewDuration=istream.readUI16();
        retVal+=2;
    }
    // anyTimeFreepreviewDuration
    {
        anyTimeFreepreviewDuration=istream.readUI16();
        retVal+=2;
    }
    // data
    {
        retVal+=data->read(istream);
    }
    // data1
    {
        retVal+=data1->read(istream);
    }
    // descriptors
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_CaDescriptor__PTR_TYPE temp;
        org_himalay_ccif_CaDescriptor__NEW( temp);
        retVal += temp->read(istream);
        descriptors.add(temp);
    }
    return retVal;
}

int programCnfMsg::read(DataInputStream&  istream ) { // throws IOException
    int retVal = 0;
    // read purchaseType
    {
        purchaseType=(istream.readUI8());
        retVal+=1;
    }
    // read purchasePrice
    {
        purchasePrice=istream.readUI16();
        retVal+=2;
    }
    // read purchaseValidation
    {
        purchaseValidation=(istream.readUI8());
        retVal+=1;
    }
    // read expirationdatee
    {
        expirationdatee=istream.readUI32() ;
        retVal+=4;
    }
    // read ProgramStartTim
    {
        ProgramStartTim=istream.readUI32() ;
        retVal+=4;
    }
    // read initialFreepreviewDuration
    {
        initialFreepreviewDuration=istream.readUI16();
        retVal+=2;
    }
    // read anyTimeFreepreviewDuration
    {
        anyTimeFreepreviewDuration=istream.readUI16();
        retVal+=2;
    }
    // read data
    {
        retVal+=data->read(istream);
    }
    // read data1
    {
        retVal+=data1->read(istream);
    }
    // read descriptors
    for (; istream.available() > 0 ; ) {
        org_himalay_ccif_CaDescriptor__PTR_TYPE temp;
        org_himalay_ccif_CaDescriptor__NEW( temp);
        retVal += temp->read(istream);
        descriptors.add(temp);
    }

    return retVal;
}

int programCnfMsg::write(DataOutputStream& ostream) { // throws IOException







    {   /** fix dependent sizes for data**/
    }
    {   /** fix dependent sizes for data1**/
    }
    {   /** fix dependent sizes for descriptors**/
    }

    int retVal= 0;
    // write purchaseType
    ostream.writeUI8(purchaseType);
    retVal +=1;
    // write purchasePrice
    ostream.writeUI16(purchasePrice);
    retVal +=2;
    // write purchaseValidation
    ostream.writeUI8(purchaseValidation);
    retVal +=1;
    // write expirationdatee
    ostream.writeUI32(expirationdatee);
    retVal +=4;
    // write ProgramStartTim
    ostream.writeUI32(ProgramStartTim);
    retVal +=4;
    // write initialFreepreviewDuration
    ostream.writeUI16(initialFreepreviewDuration);
    retVal +=2;
    // write anyTimeFreepreviewDuration
    ostream.writeUI16(anyTimeFreepreviewDuration);
    retVal +=2;
    // write data
    {
        retVal += data->write(ostream);
    }
    // write data1
    {
        retVal += data1->write(ostream);
    }
    // write descriptors
{   for (ArrayList_iterator(CaDescriptor) it=  descriptors.getIterator() ; it.hasNext();) {
            {
                retVal +=it.get()->write(ostream);
            }
        }
    }

    return retVal;
}

int programCnfMsg::dump(DumpContext& dc) { // throws IOException
    dc.indent();
    dc.getPs().print("programCnfMsg\n");
    dc.increaseIndent();
    int retVal= 0;
    // write purchaseType
    dc.indent();
    dc.getPs().print("purchaseType=");
    dc.getPs().println((long)purchaseType);
    // write purchasePrice
    dc.indent();
    dc.getPs().print("purchasePrice=");
    dc.getPs().println((long)purchasePrice);
    // write purchaseValidation
    dc.indent();
    dc.getPs().print("purchaseValidation=");
    dc.getPs().println((long)purchaseValidation);
    // write expirationdatee
    dc.indent();
    dc.getPs().print("expirationdatee=");
    dc.getPs().println((long)expirationdatee);
    // write ProgramStartTim
    dc.indent();
    dc.getPs().print("ProgramStartTim=");
    dc.getPs().println((long)ProgramStartTim);
    // write initialFreepreviewDuration
    dc.indent();
    dc.getPs().print("initialFreepreviewDuration=");
    dc.getPs().println((long)initialFreepreviewDuration);
    // write anyTimeFreepreviewDuration
    dc.indent();
    dc.getPs().print("anyTimeFreepreviewDuration=");
    dc.getPs().println((long)anyTimeFreepreviewDuration);
    // write data
    dc.indent();
    dc.getPs().print("data");
    data->dump(dc);
    // write data1
    dc.indent();
    dc.getPs().print("data1");
    data1->dump(dc);
    // write descriptors
{   for (ArrayList_iterator(CaDescriptor) it=  descriptors.getIterator() ; it.hasNext();) {
            {
                dc.indent();
                dc.getPs().println("element");
                retVal +=it.get()->dump(dc);
            }
        }
    }
    dc.decreaseIndent();
    return retVal;
}


// Getter for purchaseType
//ui8 programCnfMsg::getPurchaseType()
//{
//return this->purchaseType;
//}

// Setter for purchaseType
//void programCnfMsg::setPurchaseType(ui8 val)
//{
//this->purchaseType= val;
//}

// Getter for purchasePrice
//ui16 programCnfMsg::getPurchasePrice()
//{
//return this->purchasePrice;
//}

// Setter for purchasePrice
//void programCnfMsg::setPurchasePrice(ui16 val)
//{
//this->purchasePrice= val;
//}

// Getter for purchaseValidation
//ui8 programCnfMsg::getPurchaseValidation()
//{
//return this->purchaseValidation;
//}

// Setter for purchaseValidation
//void programCnfMsg::setPurchaseValidation(ui8 val)
//{
//this->purchaseValidation= val;
//}

// Getter for expirationdatee
//ui32 programCnfMsg::getExpirationdatee()
//{
//return this->expirationdatee;
//}

// Setter for expirationdatee
//void programCnfMsg::setExpirationdatee(ui32 val)
//{
//this->expirationdatee= val;
//}

// Getter for ProgramStartTim
//ui32 programCnfMsg::getProgramStartTim()
//{
//return this->ProgramStartTim;
//}

// Setter for ProgramStartTim
//void programCnfMsg::setProgramStartTim(ui32 val)
//{
//this->ProgramStartTim= val;
//}

// Getter for initialFreepreviewDuration
//ui16 programCnfMsg::getInitialFreepreviewDuration()
//{
//return this->initialFreepreviewDuration;
//}

// Setter for initialFreepreviewDuration
//void programCnfMsg::setInitialFreepreviewDuration(ui16 val)
//{
//this->initialFreepreviewDuration= val;
//}

// Getter for anyTimeFreepreviewDuration
//ui16 programCnfMsg::getAnyTimeFreepreviewDuration()
//{
//return this->anyTimeFreepreviewDuration;
//}

// Setter for anyTimeFreepreviewDuration
//void programCnfMsg::setAnyTimeFreepreviewDuration(ui16 val)
//{
//this->anyTimeFreepreviewDuration= val;
//}

// Getter for data
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE programCnfMsg::getData()
//{
//return this->data;
//}

// Setter for data
//void programCnfMsg::setData(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->data= val;
//}

// Getter for data1
//org_himalay_msgs_runtime_ByteArray_PTR_TYPE programCnfMsg::getData1()
//{
//return this->data1;
//}

// Setter for data1
//void programCnfMsg::setData1(org_himalay_msgs_runtime_ByteArray_PTR_TYPE val)
//{
//this->data1= val;
//}

// Getter for descriptors
//ArrayList(CaDescriptor) programCnfMsg::getDescriptors()
//{
//return this->descriptors;
//}

// Setter for descriptors
//void programCnfMsg::setDescriptors(ArrayList(CaDescriptor) val)
//{
//this->descriptors= val;
//}

int programCnfMsg::getSize() { // throws IOException
    int retVal = 0;

    return retVal;
}

#ifdef org_himalay_ccif_programCnfMsg__USE_SMART_PTR
void intrusive_ptr_add_ref(org::himalay::ccif::programCnfMsg* p) {
    p->increaseRef();
};
void intrusive_ptr_release(org::himalay::ccif::programCnfMsg* p) {
    p->decreaseRef();
};;
#endif


}
}
}

// End of code